### PR TITLE
update llama3.2 vision memory estimation

### PR DESCRIPTION
**Pull request by mxyng**
adjust estimations for mllama which has conditional graph components and a different cache shape

### CODE REVIEW

File: llm/ggml.go [hunk 1]
```diff
@@ -400,6 +403,16 @@ func (llm GGML) GraphSize(context, batch uint64) (partialOffload, fullOffload ui
...
 			)
 		}
+	case "mllama":
+		var numVisionTokens, numTiles uint64 = 1601, 4
+		if crossAttentionLayers, ok := llm.KV()["mllama.cross_attention_layers"].(*array); ok {
+			kv = 2*context*(llm.KV().BlockCount()-uint64(crossAttentionLayers.size))*(embeddingHeadsK+embeddingHeadsV)*headsKV +
+				4*2*uint64(crossAttentionLayers.size)*embeddingHeadsK*numVisionTokens*numTiles*headsKV
+		}
+
+		fullOffload = 4 * (numVisionTokens*numTiles*embedding +
```
Comment (pdevine) [inline llm/ggml.go:413]: Do you need to set `partialOffload` as well?

File: llm/ggml.go [hunk 2]
```diff
@@ -400,6 +403,16 @@ func (llm GGML) GraphSize(context, batch uint64) (partialOffload, fullOffload ui
 				4*batch*(1+2*embedding+context*(1+heads))+embedding*(6*context*headsKV/heads+embedding*9/16),
 			)
 		}
+	case "mllama":
+		var numVisionTokens, numTiles uint64 = 1601, 4
+		if crossAttentionLayers, ok := llm.KV()["mllama.cross_attention_layers"].(*array); ok {
+			kv = 2*context*(llm.KV().BlockCount()-uint64(crossAttentionLayers.size))*(embeddingHeadsK+embeddingHeadsV)*headsKV +
```
Comment (bmizerany) [inline llm/ggml.go:409]: For future readers, it would be helpful if this was broken down, and had comments per logical operation. The lengthy variable names and absence of newlines makes this hard to parse.

**Review by jmorganca — APPROVED** [review]
